subworkflows/local/protein_annotation.nf

```diff
@@ -0,0 +1,55 @@
+/*
+    RUN FUNCTIONAL CLASSIFICATION
+*/
+
+include { INTERPROSCAN_DATABASE } from '../../modules/local/interproscan_download'
+include { INTERPROSCAN          } from '../../modules/nf-core/interproscan/main'
+
+workflow PROTEIN_ANNOTATION {
+    take:
+    faas // tuple val(meta), path(PROKKA/PRODIGAL.out.faa)
+
+    main:
+    ch_versions                  = Channel.empty()
+    ch_interproscan_tsv          = Channel.empty()
+    ch_interproscan_db           = Channel.empty()
+    ch_interproscan_tsv_modified = Channel.empty()
+
+    ch_faa_for_interproscan = faas
+
+    if ( params.protein_annotation_tool == 'InterProScan') {
+
+        if ( params.protein_annotation_interproscan_db != null ) {
+            ch_interproscan_db = Channel
+                .fromPath( params.protein_annotation_interproscan_db )
+                .first()
+        } else {
+            INTERPROSCAN_DATABASE ( params.protein_annotation_interproscan_db_url )
+            ch_versions  = ch_versions.mix( INTERPROSCAN_DATABASE.out.versions )
+            ch_interproscan_db = ( INTERPROSCAN_DATABASE.out.db )
+        }
+
+        INTERPROSCAN( ch_faa_for_interproscan, ch_interproscan_db )
+        ch_versions  = ch_versions.mix( INTERPROSCAN.out.versions )
+        ch_interproscan_tsv = ch_interproscan_tsv.mix( INTERPROSCAN.out.tsv )
+
+        // Current INTERPROSCAN version 5.59_91.0 only includes 13 columns and not 15 which ampcombi expects, so we added them here
```

Comment:
Isn't this something to solve upstream on AMPcombi side? :grimacing: Is ok for now I guess, but better to have this column number check done by AMPcombi instead of pipeline level.

Reply:
And thats where updating the database version is not a good idea before testing. 
AMPcombi was developed on an updated version of InterProScan to the one available as a nf-core module. So its rather updating the module on nf-core than downgrading AMPcombi.

Reply:
Okay! Fair point if AMPcombi is tested to work with a specific interproscan version :+1: Just need to remember to edit this code when interproscan nf-core module gets updated one day. We could do it :nerd_face:

Reply:
If u want to get involved here is what they had to go through https://github.com/bioconda/bioconda-recipes/issues/21601 :)